operator/src/test/java/org/keycloak/operator/testsuite/unit/IngressLogicTest.java

```diff
@@ -72,29 +76,49 @@ protected void deleteExistingIngress() {
     }
 
     @Test
-    public void testIngressEnabledExisting() {
-        var kc = MockKeycloakIngress.build(true, true);
+    public void testIngressDisabledExisting() {
+        var kc = MockKeycloakIngress.build(false, true,
+                                                                            true);
         assertFalse(kc.reconciled());
         assertTrue(kc.deleted());
     }
 
     @Test
-    public void testIngressEnabledNotExisting() {
-        var kc = MockKeycloakIngress.build(true, false);
+    public void testIngressDisabledNotExisting() {
+        var kc = MockKeycloakIngress.build(false, false,
+                                                                            true);
         assertFalse(kc.reconciled());
         assertFalse(kc.deleted());
     }
 
     @Test
-    public void testIngressDisabledExisting() {
-        var kc = MockKeycloakIngress.build(false, true);
+    public void testIngressEnabledExisting() {
+        var kc = MockKeycloakIngress.build(true, true,
+                                                                            true);
         assertTrue(kc.reconciled());
         assertFalse(kc.deleted());
     }
 
     @Test
-    public void testIngressDisabledNotExisting() {
-        var kc = MockKeycloakIngress.build(false, false);
+    public void testIngressEnabledNotExisting() {
+        var kc = MockKeycloakIngress.build(true, false,
+                                                                            true);
+        assertTrue(kc.reconciled());
+        assertFalse(kc.deleted());
+    }
+
+    @Test
+    public void testIngressEnabledNotSpecified() {
+        var kc = MockKeycloakIngress.build(true, false,
+                                                                            false);
+        assertTrue(kc.reconciled());
+        assertFalse(kc.deleted());
+    }
+
+    @Test
+    public void testIngressDisabledNotSpecified() {
+        var kc = MockKeycloakIngress.build(false, false,
+                false);
         assertTrue(kc.reconciled());
         assertFalse(kc.deleted());
     }
```

Comment:
I think this can never happen, right? To have Ingress disabled but NOT specified.

Reply:
@vmuzikar right, removed.